### DISCUSSION TITLE
[BUG] Sync-PnPAppToTeams Authentication Method is not allowed

### DISCUSSION THREAD

**Discussion by thehorseofcourse** [pnp/powershell#2554] [General]
### Reporting an Issue or Missing Feature
I am reporting a bug

### Expected behavior 
Using `Sync-PnPAppToTeams` should deploy an App in the SharePoint App catalog to Teams

### Actual behavior 
I am currently trying to deploy an App that is in my SharePoint App catalog to Teams. I first connect to my app catalog with `Connect-PnPOnline` with an account which has global admin, SharePoint Admin and Teams Admin permissions. 

I then get the following error when running `Sync-PnPAddToTeams`

![image](https://user-images.githubusercontent.com/84247314/198973985-b641796b-dae6-476f-b475-827c669c854d.png)

### Steps to reproduce behavior
Connect-PnPOnline -Url https://contoso.sharepoint.com/sites/apps
Sync-PnPAppToTeams -Identity "AppID"

OR 
Connect-PnPOnline -Url https://contoso.sharepoint.com/sites/apps -ClientId $(SPFxclientId) -CertificatePath $certificatePath -CertificatePassword $SecurePassword  -Tenant $tenantName
Sync-PnPAppToTeams -Identity "AppID"

### What is the version of the Cmdlet module you are running?
![image](https://user-images.githubusercontent.com/84247314/198326831-14351369-c355-4510-bbb2-9fb067c6fbbc.png)

### Which operating system/environment are you running PnP PowerShell on?
- [X ] Windows

**Comment by gautamdsheth** [comment]
Hii looks like some issue with your installation, this is not a bug.

Can you maybe try to uninstall and re-install ? or maybe install it on a different machine.

**Comment by thehorseofcourse** [comment]
I've uninstalled and tried again. Get the same error. Tried on another machine. Get the same error. Colleague gets the same error.
Uninstalled using - Uninstall-Module PnP.Powershell
installed using - Install-Module -Name PnP.PowerShell
Do you have any other suggestions please as to how i can rectify this?

**Comment by asadrefai** [comment]
Hi Dave (@thehorseofcourse), 
Unfortunately I am also getting the same error. 
CC: @gautamdsheth

**Comment by gautamdsheth** [comment]
What is the exact error that you are facing ?

The one in the screenshot is "... is not recognized as a cmdlet , ... etc" . Is a generic error and has nothing to do with PnP PowerShell.

Maybe this link can help:

https://techcommunity.microsoft.com/t5/windows-powershell/the-term-is-not-recognized-as-the-name-of-a-cmdlet/m-p/1414518

**Comment by thehorseofcourse** [comment]
@gautamdsheth I have been an idiot and copied the wrong screenshot. Very sorry. I have updated with the actual problem. Sorry for the inconvenience.

**Comment by gautamdsheth** [comment]
@thehorseofcourse / @asadrefai - looks like you are using Weblogin method to authenticate. Would suggest that you change the approach and use a different authentication model for SharePoint via OAuth and an Azure AD application.

Weblogin is legacy, insecure and will be removed at some point in future.

**Comment by thehorseofcourse** [comment]
Thanks @gautamdsheth will give it a go

**Comment by thehorseofcourse** [comment]
@gautamdsheth What permissions would be needed on an app reg. To get this working, I've added a few Application for graph and sharepoint Sites.Fullcontrol, TeamsAppInstallation, AppCatalog still getting the same error. Do I need Delegated permissions instead?

**Comment by gautamdsheth** [comment]
Am moving this to discussion, I usually use delegated permissions and works fine.
If it doesn't work with application permissions, then the issue lies with the API itself in which case we can't do anything as Microsoft owns the API.

**Comment by buszi2137** [comment]
Hi, Few days ago I ocurred the same problem whith Sync-PnPAppToTeams. I have registered app in AzureAD whcich has permissions like this:
![image](https://github.com/pnp/powershell/assets/124353997/2dc543b7-f33b-4c18-b10d-27d9912cb83d)
I'm using this command to connect: connect-pnponline -Tenant {tenant}.onmicrosoft.com -CertificatePath C:\Users\Lenovo\Desktop\pnp.pfx -ClientId {AzureAD App id} -Url https://{tenant}.sharepoint.com/sites/new24
Do you know how to fix it ?
Best regards
Kamil

**Comment by DenisChimbor** [comment]
I have the same issue. In my case it doesn't work with `-Interactive` connection parameter, but works fine with `-UseWebLogin` that is not recommended.

It works:

```
Connect-PnpOnline -Url $appCatalogUrl -UseWebLogin
...
Sync-PnPAppToTeams -Identity $app.Id
```

It fails with **"Authentication Method is not allowed"** exception:
```
Connect-PnpOnline -Url $appCatalogUrl -Interactive
...
Sync-PnPAppToTeams -Identity $app.Id
```

Also it worked before, when I used _SharePointPnPPowerShellOnline_ PowerShell module. I tried to look at source code and the only difference I found is that old AppManager sets cookies if access token is empy and in new version this code is not presented.

Old AppManager (https://github.com/pnp/PnP-Sites-Core/blob/deba8844ac45fa3e481ab11c66dfa4cd85489b4d/Core/OfficeDevPnP.Core/ALM/AppManager.cs#L841):

```
if (String.IsNullOrEmpty(accessToken))
{
  handler.SetAuthenticationCookies(context);
}
```

New AppManager doesn't have this code (https://github.com/pnp/pnpframework/blob/ed3d8c5d225694d0548c05b8229b247b4e00ef33/src/lib/PnP.Framework/ALM/AppManager.cs#L803)
